### PR TITLE
gawk => 5.2.1

### PR DESCRIPTION
**Pull request by saltedcoffii**
## Description
upgrade `gawk` to latest upstream version 5.2.1

## Additional information

Works properly (tested in a container):
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

#### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/saltedcoffii/chromebrew.git CREW_TESTING_BRANCH=gawk_5.2.0 CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/gawk.rb [hunk 1]
```diff
@@ -37,21 +25,26 @@ def self.patch
...
+    end
   end
 
   def self.install
-    # Conflict with /usr/local/bin/awk from mawk package.
-    ENV['CREW_CONFLICTS_ONLY_ADVISORY'] = '1'
-    reload_constants
     system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    # Remove conflict with #{CREW_PREFIX}/bin/awk from mawk package
+    FileUtils.rm "#{CREW_DEST_PREFIX}/bin/awk"
```
Comment (uberhacker) [inline packages/gawk.rb:48]: Is gawk better than mawk?  Should we remove the conflict from malk instead?

File: packages/gawk.rb [hunk 2]
```diff
@@ -3,30 +3,18 @@
...
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gawk/5.1.1-1_i686/gawk-5.1.1-1-chromeos-i686.tpxz',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gawk/5.1.1-1_x86_64/gawk-5.1.1-1-chromeos-x86_64.tpxz'
-  })
-  binary_sha256({
-    aarch64: '0963d6d7067653da8039fc4182235d7b8b75bb4adc8e2448e122a85d00cb3d93',
-     armv7l: '0963d6d7067653da8039fc4182235d7b8b75bb4adc8e2448e122a85d00cb3d93',
-       i686: '9c6609595c2fa11e1a5e9ddda7e6a592b4040549a37915ebae929ba12fa41fea',
-     x86_64: '2734c6270be8ee586007e4389c762b956f993d7503e9cb7d863c2b675bde9e2b'
-  })
-
```
Comment (uberhacker) [inline packages/gawk.rb:25]: Any luck building binaries with @satmandu's scripts?  You should only need `tools/gl.sh` and GitLab token access to push up binaries to the repo.

File: packages/gawk.rb (hunk 1 above)
Comment (saltedcoffii) [reply packages/gawk.rb:48]: gawk has more features but its slower. we use mawk by default in buildessential for that reason, so here mawk is the default as well.

File: packages/gawk.rb (hunk 2 above)
Comment (saltedcoffii) [reply packages/gawk.rb:25]: lol hang on I need to get home to my beefy computer 😂

**Comment by satmandu** [comment]
I use `gawk` for my scripts because it allows for in-place replace. I would have no problem replacing mawk with gawk in buildessential...

**Comment by uberhacker** [comment]
@saltedcoffii: Rubocop is complaining about something petty again. :)
```
packages/gawk.rb:30:5: C: [Correctable] Layout/LeadingCommentSpace: Missing space after #.
    #system 'autoreconf -fiv'
    ^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by saltedcoffii** [comment]
that's literally what rubocop is: petty
but in actuality I think we should disable that filter: in my mind, a comment with no space in between the # and the comment is a commented out command. one with a space is an actual comment. do you agree?

**Comment by satmandu** [comment]
If you use my scripts @saltedcoffii rubocop will automatically fix that for you. :p

File: packages/gawk.rb [hunk 3]
```diff
@@ -3,30 +3,18 @@
...
-     x86_64: '2734c6270be8ee586007e4389c762b956f993d7503e9cb7d863c2b675bde9e2b'
-  })
+  source_url 'https://ftpmirror.gnu.org/gawk/gawk-5.2.0.tar.xz'
+  source_sha256 'e4ddbad1c2ef10e8e815ca80208d0162d4c983e6cca16f925e8418632d639018'
 
   depends_on 'readline' => :build
   depends_on 'ncurses'
   depends_on 'mpfr'
   depends_on 'gmp'
+  depends_on 'libsigsegv' unless ARCH == 'i686'
```
Comment (satmandu) [inline packages/gawk.rb:17]: Why isn't this used on i686?

File: packages/gawk.rb [hunk 4]
```diff
@@ -37,21 +25,23 @@ def self.patch
   end
 
   def self.build
```
Comment (satmandu) [inline packages/gawk.rb:27]: Maybe leave in `system '[ -x configure ] || autoreconf -fvi` for when we do build from tarball?

File: packages/gawk.rb (hunk 3 above)
Comment (saltedcoffii) [reply packages/gawk.rb:17]: it apparently causes segfaults. I haven't tested it myself, I just saw the comments and followed suit

File: packages/gawk.rb (hunk 4 above)
Comment (saltedcoffii) [reply packages/gawk.rb:27]: we are building from a tarball, but the git repository has configure in it anyway? I'm not sure why.

**Comment by saltedcoffii** [comment]
@satmandu @uberhacker segfaulting with or without libsigsegv on i686

**Comment by satmandu** [comment]
@saltedcoffii Do you want to open an issue upstream for the i686 build?

**Comment by satmandu** [comment]
I submitted a bug report about this upstream, as this is still an issue with 5.2.1.

**Comment by satmandu** [comment]
gawk 5.2.1 works fine w/o libsigsev on i686. Builds coming shortly.

**Comment by satmandu** [comment]
Here's what's funny:
```
chronos@cheekon-i686 /usr/local/lib/crew/packages $ lddtree `which gawk`
/usr/local/bin/gawk (interpreter => /lib/ld-linux.so.2)
    /usr/local/lib/libsigsegv.so.2 => /usr/local/lib/libsigsegv.so.2
    /usr/local/lib/libreadline.so.8 => /usr/local/lib/libreadline.so.8
        libtinfow.so.6 => /usr/local/lib/libtinfow.so.6
    /usr/local/lib/libmpfr.so.6 => /usr/local/lib/libmpfr.so.6
        libgmp.so.10 => /usr/local/lib/libgmp.so.10
    /usr/local/lib/libgmp.so.10 => /usr/local/lib/libgmp.so.10
    libdl.so.2 => /usr/local/lib/libdl.so.2
    /usr/local/lib/libm.so.6 => /usr/local/lib/libm.so.6
    libc.so.6 => /usr/local/lib/libc.so.6
chronos@cheekon-i686 /usr/local/lib/crew/packages $ gawk --version
GNU Awk 5.2.1, API 3.2, (GNU MPFR 4.1.0-p11, GNU MP 6.2.1)
Copyright (C) 1989, 1991-2022 Free Software Foundation.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see http://www.gnu.org/licenses/.
```

**Comment by satmandu** [comment]
Looks like the `--without-libsigsegv-prefix"` option fixes the build.

**Review by satmandu — APPROVED** [review]

**Review by uberhacker — APPROVED** [review]

**Pull request merged**